### PR TITLE
Correctly handle if getElementById returns null

### PR DESCRIPTION
**Pull request by hugo-vrijswijk**
I'm not sure in what cases it returns null or undefined (or if me seeing the undefined was a fever dream), but this handles both cases

### CODE REVIEW

**Review by toddburnside — APPROVED** [review]
LGTM

File: explore/src/main/scala/explore/observationtree/ObsList.scala [hunk 1]
```diff
@@ -73,14 +73,13 @@ object ObsList:
...
     Callback {
-      // 'getElementById' does not return any optional type, so coerce to js.UndefOr
-      val target: js.UndefOr[Element] =
-        dom.document.getElementById(s"obs-list-${targetObs.toString}")
-      target.foreach { t =>
-        val rect = t.getBoundingClientRect()
-        if (rect.top < 0) t.scrollIntoView()
-        if (rect.bottom > dom.window.innerHeight) t.scrollIntoView(false)
-      }
+      Option(dom.document.getElementById(s"obs-list-${targetObs.toString}"))
```
Comment (cquiroz) [inline explore/src/main/scala/explore/observationtree/ObsList.scala:76]: This got me thinking, are refs the proper react way to do this?

Comment (hugo-vrijswijk) [reply explore/src/main/scala/explore/observationtree/ObsList.scala:76]: Potentially, but with a list I don't think that would work. Or you'd have to have a single ref that you set on the last added observation, but that seems a lot more complex than a query selector

Comment (cquiroz) [reply explore/src/main/scala/explore/observationtree/ObsList.scala:76]: Right, now I recall I tried that

**Review by cquiroz — APPROVED** [review]
LGTM

**Pull request merged**